lib/ansible/module_utils/vmware.py

```diff
@@ -953,6 +953,8 @@ def get_vm(self):
             elif vms:
                 # Unique virtual machine found.
                 vm_obj = vms[0]
+        elif self.params['moid']:
+            vm_obj = VmomiSupport.templateOf('VirtualMachine')(self.params['moid'], si._stub)
```

Comment:
In this context we need to replace `si` with the pyVmomi service instance, likely in the base class of this module.  

The sample I wrote that uses this line of code for efficient lookup by moid can be found here:  https://github.com/vmware/pyvmomi/blob/master/sample/dumpjson.py#L68